src/peft/pipelines/base.py

```diff
@@ -0,0 +1,71 @@
+# coding=utf-8
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Any, Union
+
+import torch
+
+
+class BasePeftPipeline(ABC):
+    r"""
+    Base skeleton class for all PEFT pipelines. This class cannot be used as it is and should be subclassed and the
+    method `_load_model_and_processor` should be implemented.
```

Comment:
`__call__` should also be mentioned here.

Reply:
Done!